### PR TITLE
Tests for `NumTargetDriftAnalyzer`, code simplifications and documentation

### PR DESCRIPTION
**Pull request by burkovae**
- [x] Write test for current behavior
- [x] Refactor and simplify code
  - [x]   When target column is categorical, we raise a `ValueError` with explanation instead a cryptic one
- [x] Ask a bunch of questions
  - [x] Any suggestion what we should do about ` test_computing_of_correlations_between_columns_fails_for_second_data_when_columns_missing` ? (the second data is missing numerical feature from the first one). Should we handle it gracefully or raise a `ValueError`
  - [x] Any suggestion what we should do about  `test_computing_of_correlations_between_columns_fails_for_second_data_when_columns_missing_2` (column names do mismatch in both datasets). Should we handle it gracefully or raise a `ValueError`

- [x]  move to pytest

### CODE REVIEW

**Comment by Liraim** [comment]
Both cases: `test_computing_of_correlations_between_columns_fails_for_second_data_when_columns_missing` and `test_computing_of_correlations_between_columns_fails_for_second_data_when_columns_missing_2` ideally should raise some specific error about data format mismatch (eg, missing required columns, mismatch feature columns etc).

Currently it wasn't implemented (but we're planning to add such checks in future).

For now I would suggest to add generic 'test should rise some error' assertion for this tests.

**Review by Liraim — CHANGES_REQUESTED** [review]

File: evidently/analyzers/num_target_drift_analyzer.py [hunk 1]
```diff
@@ -9,47 +9,78 @@
...
-            # corr
-            ref_pred_corr = reference_data[num_feature_names + [prediction_column]].corr()[prediction_column]
-            curr_pred_corr = current_data[num_feature_names + [prediction_column]].corr()[prediction_column]
-            prediction_corr = {'reference': ref_pred_corr.to_dict(), 'current': curr_pred_corr.to_dict()}
-            result['metrics']['prediction_correlations'] = prediction_corr
+        target_metrics = _compute_correlation(reference_data, current_data, 'target',
+                                              columns.utility_columns.target, columns.num_feature_names, func)
+        prediction_metrics = _compute_correlation(reference_data, current_data, 'prediction',
+                                                  columns.utility_columns.prediction, columns.num_feature_names, func)
+        result['metrics'] = dict(**result['metrics'], **target_metrics, **prediction_metrics)
```
[the remark this answers is not in the document]
Comment (burkovae) [reply evidently/analyzers/num_target_drift_analyzer.py:84]: yes, good spotting!

File: evidently/analyzers/num_target_drift_analyzer.py [hunk 2]
```diff
@@ -9,47 +9,78 @@
 from evidently.analyzers.utils import process_columns
 
 
+def _compute_correlation(reference_data: pd.DataFrame, current_data: pd.DataFrame, prefix,
```
[the remark this answers is not in the document]
Comment (burkovae) [reply evidently/analyzers/num_target_drift_analyzer.py:12]: you mean for other parameters?

File: evidently/analyzers/test_numerical_target_drift_analyzer.py [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+    })
+
+    result = analyzer.calculate(df1, df2, ColumnMapping(target='another_target'))
+    _assert_result_structure(result)
+    assert result['metrics']['target_name'] == 'another_target'
+    assert result['metrics']['target_type'] == 'num'
+    assert result['metrics']['target_drift'] == 1
+    correlations = result['metrics']['target_correlations']
+    assert correlations['reference'] == {'another_target': 1.0}
+    assert correlations['current'], {'another_target': 1.0}
```
[the remark this answers is not in the document]
Comment (burkovae) [reply evidently/analyzers/test_numerical_target_drift_analyzer.py:55]: Awesome! Yes, that was a typo!

File: evidently/analyzers/num_target_drift_analyzer.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Liraim) [reply evidently/analyzers/num_target_drift_analyzer.py]: Yes, exactly.

**Review by Liraim — APPROVED** [review]
Looks good now.

**Pull request merged**